### PR TITLE
Stop swallowing exceptions in TracingInterceptor

### PR DESCRIPTION
**Pull request by nbroyles**
Not really sure what I was thinking here 🤦‍♂️

### CODE REVIEW

File: misk/src/main/kotlin/misk/web/interceptors/TracingInterceptor.kt [hunk 1]
```diff
@@ -29,17 +29,8 @@ internal class TracingInterceptor internal constructor(private val tracer: Trace
...
-    try {
-      scope = tracer.buildSpan("${chain.action}").startActive(true)
-      result = chain.proceed(chain.args)
-    } catch (exception: Exception) {
-      logger.warn("failed attempting to trace request", exception)
-      if (scope != null) Tags.ERROR.set(scope.span(), true)
-    } finally {
-      if (scope != null) scope.close()
-    }
+    tracer.buildSpan("${chain.action}").startActive(true).use { result = chain.proceed(chain.args) }
```
[the remark this answers is not in the document]
Comment (nbroyles) [reply misk/src/main/kotlin/misk/web/interceptors/TracingInterceptor.kt]: `use` is like try-with-resources for Kotlin and will handle any Closeable. https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/use.html

[the remark this answers is not in the document]
Comment (eeSeeGee) [reply misk/src/main/kotlin/misk/web/interceptors/TracingInterceptor.kt]: Ah, gotcha

File: misk/src/main/kotlin/misk/web/interceptors/TracingInterceptor.kt [hunk 2]
```diff
@@ -29,17 +29,8 @@ internal class TracingInterceptor internal constructor(private val tracer: Trace
...
   override fun intercept(chain: Chain): Any? {
     var result: Any? = null
-    var scope: Scope? = null
 
-    try {
-      scope = tracer.buildSpan("${chain.action}").startActive(true)
-      result = chain.proceed(chain.args)
-    } catch (exception: Exception) {
-      logger.warn("failed attempting to trace request", exception)
-      if (scope != null) Tags.ERROR.set(scope.span(), true)
```
Comment (ryanhall07) [inline misk/src/main/kotlin/misk/web/interceptors/TracingInterceptor.kt]: so what was the benefit of tagging an `ERROR`?

Comment (nbroyles) [reply misk/src/main/kotlin/misk/web/interceptors/TracingInterceptor.kt]: Going to put this back in as a part of fixing https://github.com/square/misk/issues/125

Comment (ryanhall07) [reply misk/src/main/kotlin/misk/web/interceptors/TracingInterceptor.kt]: nice

**Pull request merged**